docs/intro.md

```diff
@@ -0,0 +1,94 @@
+# Introduction to Mu-Haskell
+
+## What is Mu-Haskell?
```

Comment:
I would suggest a phrase describing the context, the trend, and the problem or difficulty which Mu-Haskell is intended to solve. 

Here is some sketches, fill as convenient: 

> Many companies have embraced Micro-Services architectures as the best way to scale up their internal software systems, separate work across different company divisions and development teams. 
> Microservices architectures also allow teams to turn an idea or bug report into a working feature of fix in production more quickly, in accordance to the agile principles.   
> However, micro-services are not without costs. Every connection between micro-services becomes now a boundary that requires one to act as a server, and the other to act as the client. Each part implementation needs to add the protocol, the codification of the data for transmission, etc.   
> Also, business logic of the application starts to spread around several code-bases, making it difficult to maintain.

Reply:
Hi Diego! Thanks, you are right, thanks for the recommendation! Unfortunately, this change goes on this https://github.com/higherkindness/mu-haskell/pull/26PR from @serras, I started this branch from his new docs branch to dogfood the new Stack template he created (templates/grpc-server.hsfiles) 😄